salt/daemons/masterapi.py

```diff
@@ -389,6 +416,8 @@ def check_autosign(self, keyid):
             return True
         if self.check_autosign_dir(keyid):
             return True
+        if self.check_autosign_grains(autosign_grains):
```

Comment:
I think this might need to be `if autosign_grains and self.check_autosign_grains(autosign_grains):` or we'll be passing `None` into that function when it's expecting a list.

Reply:
I am checking `if not autosign_grains: return False` in the function: https://github.com/saltstack/salt/pull/44324/files#diff-b63dd508d9b7489d1f22527bda3f300dR377